SMLHelper/Patchers/CraftDataPatcher.cs

```diff
@@ -88,10 +88,44 @@ internal static void Patch(Harmony harmony)
 #endif
             harmony.Patch(AccessTools.Method(typeof(CraftData), nameof(CraftData.PreparePrefabIDCache)),
                postfix: new HarmonyMethod(AccessTools.Method(typeof(CraftDataPatcher), nameof(CraftDataPrefabIDCachePostfix))));
+            PatchUtils.PatchClass(harmony);
 
             InternalLogger.Log("CraftDataPatcher is done.", LogLevel.Debug);
         }
 
+        [PatchUtils.Prefix]
+        [HarmonyPatch(typeof(CraftData), nameof(CraftData.GetTechType), new Type[] { typeof(GameObject), typeof(GameObject) }, argumentVariations: new ArgumentType[] { ArgumentType.Normal, ArgumentType.Out })]
+        private static void CraftDataGetTechTypePrefix(GameObject obj, out GameObject go, ref TechType __result)
+        {
+            CraftData.PreparePrefabIDCache();
+            Transform transform = obj.transform;
+            TechTag techTag = null;
+            PrefabIdentifier prefabIdentifier = null;
+
+            while(transform != null && !transform.TryGetComponent(out prefabIdentifier) && !transform.TryGetComponent(out techTag))
+            {
+                transform = transform.parent;
+            }
+
+            if(prefabIdentifier != null)
+            {
+                go = prefabIdentifier.gameObject;
+                __result = CraftData.entClassTechTable.GetOrDefault(prefabIdentifier.ClassId, TechType.None);
+                return;
+            }
+
+            if(techTag != null)
+            {
+                go = techTag.gameObject;
+                __result = techTag.type;
+                return;
+            }
+
+            go = null;
+            __result = TechType.None;
+            return;
+        }
+
```

Comment:
This Patch allows the game to load modded techtypes even if the caches all get deleted as it will now get the type from the classid instead of the values of the TechTag